EventHandler.java

```diff
@@ -0,0 +1,170 @@
+package SearchDemo;
+
+import com.aliyun.fc.runtime.*;
+import com.aliyun.opensearch.DocumentClient;
+import com.aliyun.opensearch.OpenSearchClient;
+import com.aliyun.opensearch.sdk.dependencies.com.google.common.collect.Maps;
+import com.aliyun.opensearch.sdk.generated.OpenSearch;
+import com.aliyun.opensearch.sdk.generated.commons.OpenSearchClientException;
+import com.aliyun.opensearch.sdk.generated.commons.OpenSearchException;
+import com.aliyun.opensearch.sdk.generated.commons.OpenSearchResult;
+import com.aliyun.oss.OSSClient;
+import com.aliyun.oss.model.OSSObject;
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+
+import java.io.*;
+import java.util.*;
+
+public class EventHandler implements StreamRequestHandler {
+
+    private static String ossEndpoint = "YourOSSEndpoint";
+    private static String openSearchAppName = "YourOpenSearchAppName";
+    private static String openSearchHost = "YourOpenSearchHost";
+    private static String openSearchTableName = "YourOpenSearchTableName";
+    private static String accessKeyId = "YourAccessKeyId";
+    private static String accessKeySecret = "YourAccessSecretId";
+    private static String docUrlFormat = "http://%s.%s/%s";
+
+    private static String[] addEventArray = {"ObjectCreated:PutObject", "ObjectCreated:PostObject"};
+    private static List<String> addEventList = Arrays.asList(addEventArray);
+    private static String[] updateEventArray = {"ObjectCreated:AppendObject"};
+    private static List<String> updateEventList = Arrays.asList(updateEventArray);
+    private static String[] deleteEventArray = {"ObjectRemoved:DeleteObject", "ObjectRemoved:DeleteObjects"};
+    private static List<String> deleteEventList = Arrays.asList(deleteEventArray);
+
+
+    @Override
+    public void handleRequest(
+            InputStream inputStream, OutputStream outputStream, Context context) throws IOException {
+
+        /*
+         * Preparation
+         * Init logger, oss client, open search document client.
+         */
+        FunctionComputeLogger fcLogger = context.getLogger();
+        OSSClient ossClient = getOSSClient(context);
+        DocumentClient documentClient = getDocumentClient();
+
+        /*
+         * Step 1
+         * Read oss event from input stream.
+         */
+        JSONObject ossEvent;
+        StringBuilder inputBuilder = new StringBuilder();
+        BufferedReader streamReader = null;
+        try {
+            streamReader = new BufferedReader(new InputStreamReader(inputStream));
+            String line;
+            while ((line = streamReader.readLine()) != null) {
+                inputBuilder.append(line);
+            }
+            fcLogger.info("Read object event success.");
+        } catch(Exception ex) {
+            fcLogger.info(ex.getMessage());
```

Comment:
If reading encountered error, should we return early here?

Reply:
Yes, we should return early. Are we suggest costumer to convert exception to IOexception and throw again?

Reply:
Yes, we should return early in catch.